yt_dlp/extractor/twitter.py

```diff
@@ -107,54 +105,34 @@ def _call_api(self, path, video_id, query={}, graphql=False):
-                            self._API_BASE + 'guest/activate.json', video_id,
-                            'Downloading guest token', data=b'', headers=headers)
-
-                        self._TOKENS[bearer_token] = guest_token_response.get('guest_token')
-                        if not self._TOKENS[bearer_token]:
-                            raise ExtractorError('Could not retrieve guest token')
-
-                    headers['x-guest-token'] = self._TOKENS[bearer_token]
-
-                try:
-                    allowed_status = {400, 403, 404} if graphql else {403}
-                    result = self._download_json(
-                        (self._GRAPHQL_API_BASE if graphql else self._API_BASE) + path,
-                        video_id, headers=headers, query=query, expected_status=allowed_status)
-
-                except ExtractorError as e:
-                    if last_error:
-                        raise last_error
-
-                    if not isinstance(e.cause, urllib.error.HTTPError) or e.cause.code != 404:
-                        raise
-
-                    last_error = e
-                    self.report_warning(
-                        'Twitter API gave 404 response, retrying with deprecated auth token. '
-                        'Only one media item can be extracted')
-                    break  # continue outer loop with next bearer_token
-
-                if result.get('errors'):
-                    errors = traverse_obj(result, ('errors', ..., 'message'), expected_type=str)
-                    if first_attempt and any('bad guest token' in error.lower() for error in errors):
-                        self.to_screen('Guest token has expired. Refreshing guest token')
-                        self._TOKENS[bearer_token] = None
-                        continue
+        for first_attempt in (True, False):
+            if not self.is_logged_in:
+                if not self._token:
+                    headers.pop('x-guest-token', None)
+                    self._token = traverse_obj(self._download_json(
+                        self._API_BASE + 'guest/activate.json', video_id,
+                        'Downloading guest token', data=b'', headers=headers), 'guest_token')
+                    if not self._token:
+                        raise ExtractorError('Could not retrieve guest token')
+                headers['x-guest-token'] = self._token
+
+            allowed_status = {400, 403, 404} if graphql else {403}
+            result = self._download_json(
+                (self._GRAPHQL_API_BASE if graphql else self._API_BASE) + path,
+                video_id, headers=headers, query=query, expected_status=allowed_status,
+                note=f'Downloading {"GraphQL" if graphql else "legacy API"} JSON')
+
+            if result.get('errors'):
+                errors = traverse_obj(result, ('errors', ..., 'message', {str}))
+                if first_attempt and any('bad guest token' in error.lower() for error in errors):
```

Comment:
```suggestion
                if first_attempt and not self.is_logged_in and any('bad guest token' in error.lower() for error in errors):
```
correct? Otherwise it may unnecessarily try twice when logged in.

Reply:
it wouldn't hurt to add, but I don't think it's necessary as we are never passing a guest token when logged in